tests/e2e-net-test.sh

```diff
@@ -0,0 +1,199 @@
+#!/bin/bash
```

Comment:
I assume this is mostly a duplicate of the original `kerneltest.sh` right? Could you elaborate why you chose to duplicate it instead of reusing the former? I think it would be beneficial if we could use the former as a reusable basis.

Reply:
Well there are differences in the two, the best we could do is to make a third file and use it as a library (with functions and common code) for both. Can I add this as a TODO instead of doing it now ? I promise I'll do it (but I wanted to check test results during the subsequent (of this one) PRs and adjust things right if needed first.